### DISCUSSION TITLE
Texture-packer files detected as PLSQL files

### DISCUSSION THREAD

**Discussion by rewinfrey** [github-linguist/linguist#5239] [Classification] [answered]
A user [reported on twitter](https://twitter.com/paytonrules/status/1366120593985327106?s=20) that [Texture-Packer files](https://www.codeandweb.com/texturepacker) are detected as [`PLSQL`](https://www.oracle.com/database/technologies/appdev/plsql.html), and throws off the language stats for their repo. This is due to both `Texture-Packer` and `PLSQL` both using the `.tps` file extension. Today, the `.tps` file extension is [detected as `PLSQL` by linguist](https://github.com/github/linguist/blob/b2834449f1dd9dd9b76652092bad9fc593f687d3/lib/linguist/languages.yml#L4015).

I wonder how feasible it would be to add a heuristic to detect `PLSQL` files from `Texture-Packer` files? Because `Texture-Packer` is a utility to generate artifacts and not a programming language, it seems like the appropriate thing would be to not classify `.tps` files generated by `Texture-Packer` if possible. Curious what we can do to prevent this false classification?

**Comment by lildude** [comment, marked as answer]
We resolve these kinds of problems by adding support for the mis-classified language, assuming it is sufficiently popular on GitHub (in use in more than 200 unique user/repo combinations). This can be done for non-programming languages too by classifying the language as "data" which will stop it from appearing in the stats by default but will also ensure the files are correctly classified.

The CONTRIBUTING.md file has all the deets on how to add a new language.
